File: CS/ShowNewObjectExample.Web/ApplicationCode/WebApplication.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Xpo;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using DevExpress.ExpressApp.Web;
using DevExpress.ExpressApp.Security;

namespace ShowNewObjectExample.Web {
    public partial class ShowNewObjectExampleAspNetApplication : WebApplication {
        protected override void CreateDefaultObjectSpaceProvider(CreateCustomObjectSpaceProviderEventArgs args) {
            args.ObjectSpaceProvider = new XPObjectSpaceProviderThreadSafe(args.ConnectionString, args.Connection);
        }
        private DevExpress.ExpressApp.SystemModule.SystemModule module1;
        private DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule module2;
        private ShowNewObjectExample.Module.ShowNewObjectExampleModule module3;
        //private ShowNewObjectExample.Module.Web.ShowNewObjectExampleAspNetModule module4;
        private DevExpress.ExpressApp.Security.SecurityModule securityModule1;
        private DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule module6;
        private DevExpress.ExpressApp.Security.SecuritySimple securitySimple2;
        private DevExpress.ExpressApp.Security.AuthenticationStandard authenticationStandard1;
        private DevExpress.ExpressApp.ScriptRecorder.Web.ScriptRecorderAspNetModule scriptRecorderAspNetModule1;
        private DevExpress.ExpressApp.ScriptRecorder.ScriptRecorderModuleBase scriptRecorderModuleBase1;
        private DevExpress.ExpressApp.Validation.ValidationModule module5;

        public ShowNewObjectExampleAspNetApplication() {
            InitializeComponent();
        }

        private void ShowNewObjectExampleAspNetApplication_DatabaseVersionMismatch(object sender, DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs e) {
			e.Updater.Update();
			e.Handled = true;
        }

        private void InitializeComponent() {
            this.module1 = new DevExpress.ExpressApp.SystemModule.SystemModule();
            this.module2 = new DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule();
            this.module3 = new ShowNewObjectExample.Module.ShowNewObjectExampleModule();
            this.module5 = new DevExpress.ExpressApp.Validation.ValidationModule();
            this.module6 = new DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule();
            this.securityModule1 = new DevExpress.ExpressApp.Security.SecurityModule();
            this.securitySimple2 = new DevExpress.ExpressApp.Security.SecuritySimple();
            this.authenticationStandard1 = new DevExpress.ExpressApp.Security.AuthenticationStandard();
            this.scriptRecorderAspNetModule1 = new DevExpress.ExpressApp.ScriptRecorder.Web.ScriptRecorderAspNetModule();
            this.scriptRecorderModuleBase1 = new DevExpress.ExpressApp.ScriptRecorder.ScriptRecorderModuleBase();
            ((System.ComponentModel.ISupportInitialize)(this)).BeginInit();
            // 
            // module5
            // 
            this.module5.AllowValidationDetailsAccess = true;
            // 
            // securitySimple2
            // 
            this.securitySimple2.Authentication = this.authenticationStandard1;
            this.securitySimple2.UserType = typeof(DevExpress.Persistent.BaseImpl.SimpleUser);
            // 
            // authenticationStandard1
            // 
            this.authenticationStandard1.LogonParametersType = typeof(DevExpress.ExpressApp.Security.AuthenticationStandardLogonParameters);
            // 
            // ShowNewObjectExampleAspNetApplication
            // 
            this.ApplicationName = "ShowNewObjectExample";
            this.Modules.Add(this.module1);
            this.Modules.Add(this.module2);
            this.Modules.Add(this.module6);
            this.Modules.Add(this.module3);
            this.Modules.Add(this.module5);
            this.Modules.Add(this.securityModule1);
            this.Modules.Add(this.scriptRecorderModuleBase1);
            this.Modules.Add(this.scriptRecorderAspNetModule1);
            this.Security = this.securitySimple2;
            this.LastLogonParametersRead += new System.EventHandler<DevExpress.ExpressApp.LastLogonParametersReadEventArgs>(this.ShowNewObjectExampleAspNetApplication_LastLogonParametersRead);
            this.DatabaseVersionMismatch += new System.EventHandler<DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs>(this.ShowNewObjectExampleAspNetApplication_DatabaseVersionMismatch);
            ((System.ComponentModel.ISupportInitialize)(this)).EndInit();

        }

        private void ShowNewObjectExampleAspNetApplication_LastLogonParametersRead(object sender, LastLogonParametersReadEventArgs e) {
            // Just to read demo user name for logon.
            AuthenticationStandardLogonParameters logonParameters = e.LogonObject as AuthenticationStandardLogonParameters;
            if (logonParameters != null) {
                if (String.IsNullOrEmpty(logonParameters.UserName)) {
                    logonParameters.UserName = "Sam";
                }
            }
        }
    }
}

File: VB/ShowNewObjectExample.Web/ApplicationCode/WebApplication.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Xpo
Imports System.ComponentModel
Imports DevExpress.ExpressApp.Web
Imports DevExpress.ExpressApp.Security

Namespace ShowNewObjectExample.Web
    Partial Public Class ShowNewObjectExampleAspNetApplication
        Inherits WebApplication

        Protected Overrides Sub CreateDefaultObjectSpaceProvider(ByVal args As CreateCustomObjectSpaceProviderEventArgs)
            args.ObjectSpaceProvider = New XPObjectSpaceProviderThreadSafe(args.ConnectionString, args.Connection)
        End Sub
        Private module1 As DevExpress.ExpressApp.SystemModule.SystemModule
        Private module2 As DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule
        Private module3 As ShowNewObjectExample.Module.ShowNewObjectExampleModule
        'private ShowNewObjectExample.Module.Web.ShowNewObjectExampleAspNetModule module4;
        Private securityModule1 As DevExpress.ExpressApp.Security.SecurityModule
        Private module6 As DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule
        Private securitySimple2 As DevExpress.ExpressApp.Security.SecuritySimple
        Private authenticationStandard1 As DevExpress.ExpressApp.Security.AuthenticationStandard
        Private module5 As DevExpress.ExpressApp.Validation.ValidationModule

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub ShowNewObjectExampleAspNetApplication_DatabaseVersionMismatch(ByVal sender As Object, ByVal e As DevExpress.ExpressApp.DatabaseVersionMismatchEventArgs) Handles MyBase.DatabaseVersionMismatch
            e.Updater.Update()
            e.Handled = True
        End Sub

        Private Sub InitializeComponent()
            Me.module1 = New DevExpress.ExpressApp.SystemModule.SystemModule
            Me.module2 = New DevExpress.ExpressApp.Web.SystemModule.SystemAspNetModule
            Me.module3 = New ShowNewObjectExample.[Module].ShowNewObjectExampleModule
            Me.module5 = New DevExpress.ExpressApp.Validation.ValidationModule
            Me.module6 = New DevExpress.ExpressApp.Objects.BusinessClassLibraryCustomizationModule
            Me.securityModule1 = New DevExpress.ExpressApp.Security.SecurityModule
            Me.securitySimple2 = New DevExpress.ExpressApp.Security.SecuritySimple
            Me.authenticationStandard1 = New DevExpress.ExpressApp.Security.AuthenticationStandard
            CType(Me, System.ComponentModel.ISupportInitialize).BeginInit()
            '
            'module5
            '
            Me.module5.AllowValidationDetailsAccess = True
            '
            'securitySimple2
            '
            Me.securitySimple2.Authentication = Me.authenticationStandard1
            Me.securitySimple2.UserType = GetType(DevExpress.Persistent.BaseImpl.SimpleUser)
            '
            'authenticationStandard1
            '
            Me.authenticationStandard1.LogonParametersType = GetType(DevExpress.ExpressApp.Security.AuthenticationStandardLogonParameters)
            '
            'ShowNewObjectExampleAspNetApplication
            '
            Me.ApplicationName = "ShowNewObjectExample"
            Me.Modules.Add(Me.module1)
            Me.Modules.Add(Me.module2)
            Me.Modules.Add(Me.module6)
            Me.Modules.Add(Me.module3)
            Me.Modules.Add(Me.module5)
            Me.Modules.Add(Me.securityModule1)
            Me.Security = Me.securitySimple2
            CType(Me, System.ComponentModel.ISupportInitialize).EndInit()

        End Sub

        Private Sub ShowNewObjectExampleAspNetApplication_LastLogonParametersRead(ByVal sender As System.Object, ByVal e As DevExpress.ExpressApp.LastLogonParametersReadEventArgs) Handles MyBase.LastLogonParametersRead
            ' Just to read demo user name for logon.
            Dim logonParameters As AuthenticationStandardLogonParameters = TryCast(e.LogonObject, AuthenticationStandardLogonParameters)
            If logonParameters IsNot Nothing Then
                If String.IsNullOrEmpty(logonParameters.UserName) Then
                    logonParameters.UserName = "Sam"
                End If
            End If
        End Sub
    End Class
End Namespace
